### DISCUSSION TITLE
How to disable printing root matcher with flow?

### DISCUSSION THREAD

**Discussion by j4vaovo** [projectdiscovery/nuclei#4614] [Q&A] [answered]
```console
$ nuclei -t a.yaml -u https://www.baidu.com

[aaa] [http] [info] https://www.baidu.com/
```

```console
$ nuclei -t a.yaml -u https://www.baidu.com -td

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v3.1.4

		projectdiscovery.io


Listing available v9.7.2 nuclei templates for /root/nuclei-templates
Template: /tmp/a.yaml

id: aaa

info:
  name: aaa
  author: aaa
  severity: info


flow: http(1) && http(2)

http:
  - raw:
      - |
        GET / HTTP/1.1
        Host: {{Hostname}}        

    matchers:
      - type: word
        words:
          - 'baidu'

  - raw:
      - |
        GET /robots.txt HTTP/1.1
        Host: {{Hostname}}        

    matchers:
      - type: word
        words:
          - 'aaaabaaaaabbccc'
````

http(1) && http(2) 
@tarunKoyalwar

**Comment by tarunKoyalwar** [comment, marked as answer]
@j4vaovo  thank you for flagging this, this happens because change in behaviour introduced in  v3.1.4 

Earlier we only printed last result if template contains two request with matchers by ignoring previous ones but this caused issue with flow when we were using for loop (see: https://github.com/projectdiscovery/nuclei/pull/4582) 

so starting from Nuclei v3.1.4 you need to add `internal: true` to mark a request as internal. You can refer docs for more info on this https://docs.projectdiscovery.io/templates/protocols/flow#skip-internal-matchers-in-multiprotocol-flow-templates
https://docs.projectdiscovery.io/templates/reference/matchers#internal-matchers
